writers/streamingbatchwriter/streamingbatchwriter.go

```diff
@@ -441,6 +456,11 @@ func (s *streamingWorkerManager[T]) run(ctx context.Context, wg *sync.WaitGroup,
 				ticker.Reset(s.batchTimeout)
 			}
 			done <- true
+		case err := <-outputCh:
+			if err != nil {
+				s.errCh <- fmt.Errorf("handler failed on %s: %w", tableName, err)
+				return
```

Comment:
This `return` statement is my actual contribution apart from renaming variables and updating tests, oh and better handling of error type panics.